test/unit/domain/wallets/payment-input-validator.spec.ts

```diff
@@ -0,0 +1,146 @@
+import { WalletCurrency, WalletType, PaymentInputValidator } from "@domain/wallets"
+import { AccountLevel, AccountStatus } from "@domain/accounts"
+
+describe("PaymentInputValidator", () => {
+  const dummySenderWallet: Wallet = {
+    id: "senderWalletId" as WalletId,
+    accountId: "senderAccountId" as AccountId,
+    type: WalletType.Checking,
+    currency: WalletCurrency.Btc,
+    onChainAddressIdentifiers: [],
+    onChainAddresses: () => [],
+  }
+  const dummyRecipientWallet: Wallet = {
+    id: "recipientWalletId" as WalletId,
+    accountId: "recipientAccountId" as AccountId,
+    type: WalletType.Checking,
+    currency: WalletCurrency.Btc,
+    onChainAddressIdentifiers: [],
+    onChainAddresses: () => [],
+  }
+  const dummyAccount: Account = {
+    id: "senderAccountId" as AccountId,
+    createdAt: new Date(),
+    username: "username" as Username,
+    defaultWalletId: "senderWalletId" as WalletId,
+    ownerId: "ownerId" as UserId,
+    depositFeeRatio: 0 as DepositFeeRatio,
+    withdrawFee: 0 as WithdrawFee,
+    level: AccountLevel.One,
+    status: AccountStatus.Active,
+    title: "" as BusinessMapTitle,
+    coordinates: {
+      latitude: 0,
+      longitude: 0,
+    },
+    contacts: [],
+  }
+
+  it("returns the correct types when everything is valid", async () => {
+    const getWalletFn: PaymentInputValidatorConfig = (walletId: WalletId) => {
+      const wallet = {
+        senderWalletId: dummySenderWallet,
+        recipientWalletId: dummyRecipientWallet,
+      }[walletId]
```

Comment:
I don't understand what this function is doing. I understand that getWalletFn take a `WalletId` and return a `Wallet`. I'm not sure why there is a notion of `senderWalletId` and `recipientWalletId` here (the wording would apply we're getting 2 wallets and applying it in a context of a payment)